contracts/src/TeleporterTokenSource.sol

```diff
@@ -438,6 +470,18 @@ abstract contract TeleporterTokenSource is
             uint256 sourceAmount =
                 _processReceivedTransfer(sourceBlockchainID, originSenderAddress, payload.amount);
 
+            DestinationBridgeSettings memory originSettings =
```

Comment:
I'd lean towards separately `_processReceivedTransfer` to `_processSingleTransfer` and `_processMultihopTransfer`. That way for multihop processes don't need to do two state lookups for the destination bridge settings, and also encapsulates the removing of token scale for fees, making it more obvious it's needed for multihops